### PR TITLE
chore: Convert spa/hashchange* browser

### PR DESCRIPTION
**Pull request by cwli24**
Please add a one-paragraph summary here, suitable for a release notes description. This will help with documentation.
---

### Related Issue(s)

https://new-relic.atlassian.net/browse/NR-138439
https://new-relic.atlassian.net/browse/NR-138500

### CODE REVIEW

File: tests/specs/spa/capture-interactions.e2e.js [hunk 1]
```diff
@@ -61,4 +63,74 @@ describe('spa captures interaction when', () => {
...
+        expect.objectContaining({
+          type: 'customTracer',
+          name: 'onPopstate',
+          children: []
+        })
+      ]
+    }))
+  })
+
+  it('hashchange in 2nd event listener', async () => {
```
Comment (patrickhousley) [inline tests/specs/spa/capture-interactions.e2e.js:106]: I think this name could be a little more clear. I am assuming that the purpose of this test is that the interaction is counting the window addEventListener callback in the interaction created by the click???

File: tests/specs/spa/capture-interactions.e2e.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
+import { notIE } from '../../../tools/browser-matcher/common-matchers.mjs'
+
 describe('spa captures interaction when', () => {
-  it('hashchange fires after finish', async () => {
+  it('hashchange fires after XHR loads', async () => {
```
Comment (patrickhousley) [inline tests/specs/spa/capture-interactions.e2e.js:4]: I know you only changed the name here but this test is sort of unnecessary and somehow expansive in what its testing at the same time. I don't think we need the XHR part of this test and could split the remaining into two parts: testing the interaction api and simply testing that the spa feature captures an interaction when the hash changes (separate from anything else).

**Review by patrickhousley — COMMENTED** [review]
The `capture-interactions.e2e.js` and `harvesting.e2e.js` should also probably be combined in the `spa` directory, maybe into a file named `index.e2e.js`. Having spec files with just a few tests is more expensive unless there is a clear distinction in what is being tested...like splitting out a spec file dedicated to the interaction api.

File: tests/specs/spa/capture-interactions.e2e.js [hunk 3]
```diff
@@ -61,4 +63,74 @@ describe('spa captures interaction when', () => {
       ]
     }))
   })
+
+  it.withBrowsersMatching(notIE)('hashchange is followed by a popstate', async () => {
```
[the remark this answers is not in the document]
Comment (metal-messiah) [reply tests/specs/spa/capture-interactions.e2e.js:67]: Group decided to keep these tests as 1:1 for conversion purposes and revisit usefulness another time

File: tests/specs/spa/capture-interactions.e2e.js (hunk 2 above)
Comment (metal-messiah) [reply tests/specs/spa/capture-interactions.e2e.js:4]: Group decided to keep these tests as 1:1 for conversion purposes and revisit usefulness another time

File: tests/specs/spa/capture-interactions.e2e.js (hunk 1 above)
Comment (metal-messiah) [reply tests/specs/spa/capture-interactions.e2e.js:106]:
```suggestion
  it('Spa does not prevent the bubbling of events', async () => {
```

**Review by patrickhousley — APPROVED** [review]

**Pull request merged**